samples/create-symbol-styles-from-web-styles/src/main/java/com/esri/arcgismaps/sample/createsymbolstylesfromwebstyles/components/CreateSymbolStylesFromWebStylesViewModel.kt

```diff
@@ -0,0 +1,184 @@
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class CreateSymbolStylesFromWebStylesViewModel(application: Application) : AndroidViewModel(application) {
+
+    val mapViewProxy = MapViewProxy()
+
+    // Create a unique value renderer
+    private val uniqueValueRenderer = UniqueValueRenderer().apply {
+        // Add the name of a field from the feature layer data that symbols will be mapped to
+        fieldNames.add("cat2")
+    }
+
+    // Create a feature layer from a service
+    val featureLayer =
+        FeatureLayer.createWithFeatureTable(ServiceFeatureTable("http://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/LA_County_Points_of_Interest/FeatureServer/0"))
+            .apply {
+                // Set the unique value renderer on the feature layer
+                renderer = uniqueValueRenderer
+            }
+
+    /**
+     * Only scale symbols if map scale greater than or equal to 80,000
+     */
+    fun onMapScaleChanged(scale: Double) {
+        featureLayer.scaleSymbols = scale >= 80000
+    }
+
+    // Create a map with the light gray basemap style
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISLightGray).apply {
+        // Set a viewpoint
+        initialViewpoint = Viewpoint(34.28301, -118.44186, 7000.0)
+        // Set a reference scale on the map for controlling symbol size
+        referenceScale = 100000.0
+        // Add the feature layer to the map's operational layers
+        operationalLayers.add(featureLayer)
+    }
+
+    // Create a symbol style from a web style. ArcGIS Online is used as the default portal when null is passed as the
+    // portal parameter
+    private val symbolStyle = SymbolStyle.createWithStyleNameAndPortal("Esri2DPointSymbolsStyle", null)
+
+    // Create a list of the required symbol names in the web style
+    private val symbolNames = listOf(
+        "atm",
+        "beach",
+        "campground",
+        "city-hall",
+        "hospital",
+        "library",
+        "park",
+        "place-of-worship",
+        "police-station",
+        "post-office",
+        "school",
+        "trail"
+    )
+
+    // Create a flow to hold the symbol names and icons
+    private val _symbolNamesAndIconsFlow = MutableStateFlow<List<Pair<String, BitmapDrawable>>>(emptyList())
+    val symbolNamesAndIconsFlow = _symbolNamesAndIconsFlow.asStateFlow()
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        // Load the symbols into a list
+        viewModelScope.launch {
+            _symbolNamesAndIconsFlow.value = createUniqueValueSymbols()
+        }
+
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map", error.message.toString()
+                )
+            }
+        }
```

Comment:
Instead of 2 scopes being launched, it would be better if the `createUniqueValueSymbols` is called once the map is loaded. 
```suggestion
        viewModelScope.launch {
            arcGISMap.load().onFailure { error ->
                messageDialogVM.showMessageDialog(
                    "Failed to load map", error.message.toString()
                )
            }

            // Once map is loaded, create unique value symbols
            _symbolNamesAndIconsFlow.value = createUniqueValueSymbols()
        }
```